src/data/markdown/docs/02 javascript api/11 jslib/01 httpx.md

```diff
@@ -0,0 +1,102 @@
+---
+title: "httpx"
+excerpt: "httpx is a wrapper library around the native k6 http module"
+---
+
+The `httpx` module is an external JavaScript library that wraps around the native [k6/http](/javascript-api/k6-http) module. 
+It's a http client with features that are not yet available in the native module.
+ - ability to set set http options globally (such as timeout)
+ - ability to set default tags and headers that will be used for all requests
+ - more user-friendly arguments to request functions (get, post, put take the same arguments)
+
+
+
+httpx module integrates well with functional library. 
+
+
+<Blockquote mod='warning'>
+
+#### This library is in active development
+
+This library is stable enough to be useful, but pay attention to the new versions released in jslib.k6.io. 
+
+This documentation is for the last version only. If you discover that some of the code below does not work, it most likely means that you are using an older version.
+
+</Blockquote>
+
+
+### Example
+
+<CodeGroup labels={["httpx session used together with functional library"]}>
+
+```javascript
+import { test } from 'https://jslib.k6.io/functional/0.0.1/index.js';
+import { Httpx } from 'https://jslib.k6.io/httpx/0.0.3/index.js';
```

Comment:
why not `httpx` ?

Reply:
Capitalization of class names is pretty standard in JS, so from that perspective it makes sense. However, as we already have a lot of APIs which are all lowercased, maybe lowercasing would give it a more unified look.

Reply:
`HTTPx` ? 

 I mean ... not capitalizing the whole acronym is a pretty big no no ... anywhere ... I think ?

Reply:
No, that's mainly an issue in Go. In JS it's not only fine but even encouraged to capitalize it just the way @sniku did. :)

Reply:
I ... don't agree with this, based on the mdn web docs: 
examples: 
* [ignoreBOM](https://developer.mozilla.org/en-US/docs/Web/API/TextDecoderStream/ignoreBOM)
* [URL](https://developer.mozilla.org/en-US/docs/Web/API/URL_API) and [createObjectURL](https://developer.mozilla.org/en-US/docs/Web/API/URL/createObjectURL) , [toJSON](https://developer.mozilla.org/en-US/docs/Web/API/URL/toJSON)
* [CSSPositionValue](https://developer.mozilla.org/en-US/docs/Web/API/CSSPositionValue) and all the other things starting with CSS 
* everything [here](https://developer.mozilla.org/en-US/docs/Web/API#interfaces) that starts with an acronym has the acronym uppercase 

Of course, there is also everybody's favorite [XMLHttpRequest](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest) which has the second acronym in not uppercase as it needs to separate it ... but we aren't in this case so :P

Reply:
To my knowledge the Ecmascript standard does not concern itself with how acronyms are written.

The closest I come to some kind of trace as to why it's XMLHttp and not XmlHttp or XMLHTTP is a style-guide by Microsoft that makes a difference between long (>3) and short (<= 3) acronyms where the short ones should be uppercase while the long ones should be capitalized. 

Google's style guide seem to propose something similar: https://google.github.io/styleguide/jsguide.html#naming

Reply:
I will create a repo for `httpx` and `functional` so issues like this can be discussed inside the repo. 
Maybe even we should have a template repo for jslib libraries developed for k6? :thinking: What do you think @simskij ?

Reply:
That could definitely be interesting! What would you envision that template to contain?